### PR TITLE
cp: Implement --parents

### LINKED ISSUE #1760: cp: implement --parents
GNU man pages says:
```
       --parents
              use full source file name under DIRECTORY
```

Code is in 
https://github.com/uutils/coreutils/blob/8337179454d0c952746a980ee37a32340f510a46/src/uu/cp/src/cp.rs

Tests in
https://github.com/uutils/coreutils/blob/8337179454d0c952746a980ee37a32340f510a46/tests/by-util/test_cp.rs

### PR DESCRIPTION
**Pull request by siyopao**
Closes #1760.

### CODE REVIEW

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -497,6 +497,38 @@ fn test_cp_strip_trailing_slashes() {
...
+#[test]
+fn test_cp_parents() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    let result = ucmd
+        .arg("--parents")
+        .arg(TEST_COPY_FROM_FOLDER_FILE)
+        .arg(TEST_COPY_TO_FOLDER)
+        .run();
+
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:509]: please run rustfmt on the test too

Comment (siyopao) [reply tests/by-util/test_cp.rs:509]: Oops, sorry about that. I didn't realize `cargo fmt` didn't pick it up. Done. Thanks!

Comment (sylvestre) [reply tests/by-util/test_cp.rs:509]: Oh, no worries :)

**Comment by sylvestre** [comment]
Looks like `cp --parent` is also a thing, could you please alias it?

https://github.com/coreutils/coreutils/blob/master/tests/cp/parent-perm.sh#L35

File: tests/by-util/test_cp.rs [hunk 2]
```diff
@@ -497,6 +497,45 @@ fn test_cp_strip_trailing_slashes() {
     assert_eq!(at.read(TEST_HELLO_WORLD_DEST), "Hello, World!\n");
 }
 
+#[test]
+fn test_cp_parents() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    let result = ucmd
+        .arg("--parents")
+        .arg(TEST_COPY_FROM_FOLDER_FILE)
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:506]: sorry but could you please add a new test when copying two files into a directory with this option?
thanks

Comment (siyopao) [reply tests/by-util/test_cp.rs:506]: No problem! I added a test that copied a file in a directory and one in the pwd. Would it be a better idea to touch a couple of files in a couple of different directories and test copying them to a new directory? I suppose in that case we would just check it the copy was successful?

Thanks for the reviews!

**Pull request merged**